### PR TITLE
Reorg test

### PR DESCRIPTION
**Pull request by teams2ua**
Add test about synchronization reorg.

### CODE REVIEW

File: core/test/integration/synchronization/ethereum_synchronization.cpp [hunk 1]
```diff
@@ -268,3 +270,116 @@ TEST_F(EthereumLikeWalletSynchronization, XpubETCSynchronization) {
     }
 }
 
+std::pair<std::shared_ptr<LambdaEventReceiver>, ledger::core::Future<bool>>
+createSyncReceiver() {
```
Comment (phaazon) [inline core/test/integration/synchronization/ethereum_synchronization.cpp]: Why not putting that on the same line as above?

File: core/test/lib/libledger-test/FakeUrlConnection.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#include "FakeUrlConnection.hpp"
+#include "api/HttpReadBodyResult.hpp"
+#include "utils/optional.hpp"
+#include "api/Error.hpp"
+
+namespace ledger {
+namespace core {
```
Comment (phaazon) [inline core/test/lib/libledger-test/FakeUrlConnection.cpp]: That namespace indentation is not the same one as you did with other files, where you increment the indent level. You can stick to one style, please? :)

File: core/test/integration/synchronization/ethereum_synchronization.cpp [hunk 3]
```diff
@@ -268,3 +270,116 @@ TEST_F(EthereumLikeWalletSynchronization, XpubETCSynchronization) {
...
+            ],\
+            \"block\": {\
+                \"hash\": \"0x0160a3b2b762ec42f5a1134ac260a21c98f377e153d332efb6e20534abb37f78\",\
+                \"height\": 5752778,\
+                \"time\": \"2018-06-08T10:19:10Z\"\
+            }\
+        }\
+    ]\
+}")
+                    },
```
Comment (phaazon) [inline core/test/integration/synchronization/ethereum_synchronization.cpp]: I don’t get that indentation neither.

Comment (phaazon) [reply core/test/integration/synchronization/ethereum_synchronization.cpp]: Oh I actually get it. Can you indent that line up to the same level of the rest? It’s JSON, the parser won’t care.

File: core/test/lib/libledger-test/FakeHttpClient.cpp [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#include "FakeHttpClient.hpp"
+#include "api/HttpRequest.hpp"
+#include <memory>
+#include "api/Error.hpp"
+
+namespace ledger {
+namespace core {
+namespace test {
```
Comment (phaazon) [inline core/test/lib/libledger-test/FakeHttpClient.cpp]: Indentation.

File: core/test/lib/libledger-test/FakeHttpClient.hpp [hunk 5]
```diff
@@ -0,0 +1,20 @@
+#pragma once
+#include "api/HttpClient.hpp"
+#include <unordered_map>
+#include "FakeUrlConnection.hpp"
+
+namespace ledger {
+namespace core {
+namespace test {
```
Comment (phaazon) [inline core/test/lib/libledger-test/FakeHttpClient.hpp]: Ditto.

File: core/test/lib/libledger-test/FakeUrlConnection.cpp [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        data.body = responseData;
+        data.statusCode = 200;
+        data.statusText = "OK";
+        return std::make_shared<FakeUrlConnection>(data);
+    }
+
+    FakeUrlConnection::FakeUrlConnection(const UrlConnectionData& data)
+        : _data(data) {};
+
+    int32_t FakeUrlConnection::getStatusCode() {
```
Comment (phaazon) [inline core/test/lib/libledger-test/FakeUrlConnection.cpp]: Do you really need all that code for tests? If so, why not make that function `const`?

Comment (teams2ua) [reply core/test/lib/libledger-test/FakeUrlConnection.cpp]: This is implementation of the interface `api::HttpUrlConnection`, so have not a lot of choises

File: core/test/lib/libledger-test/FakeHttpClient.hpp (hunk 5 above)
Comment (teams2ua) [reply core/test/lib/libledger-test/FakeHttpClient.hpp]: Done

File: core/test/lib/libledger-test/FakeHttpClient.cpp (hunk 4 above)
Comment (teams2ua) [reply core/test/lib/libledger-test/FakeHttpClient.cpp]: Done

File: core/test/integration/synchronization/ethereum_synchronization.cpp (hunk 1 above)
Comment (teams2ua) [reply core/test/integration/synchronization/ethereum_synchronization.cpp]: Done

File: core/test/integration/synchronization/ethereum_synchronization.cpp (hunk 3 above)
Comment (teams2ua) [reply core/test/integration/synchronization/ethereum_synchronization.cpp]: Change it.

File: core/test/lib/libledger-test/FakeUrlConnection.cpp (hunk 2 above)
Comment (teams2ua) [reply core/test/lib/libledger-test/FakeUrlConnection.cpp]: Done

**Comment by phaazon** [comment]
Thanks for taking the time to change the code! Approved!

**Pull request merged**